build-logic/convention-plugins/src/main/kotlin/social.androiddev.library.gradle.kts

```diff
@@ -0,0 +1,65 @@
+plugins {
+    id("social.androiddev.library.common")
+}
+
+val libs = the<org.gradle.accessors.dm.LibrariesForLibs>()
+
+kotlin {
+    jvm("desktop")
+    android()
+    iosX64()
+    iosArm64()
+    iosSimulatorArm64()
+
+    sourceSets {
+        // shared
+        val commonMain by getting
+
+        // iOS
+        val iosX64Main by getting
+        val iosArm64Main by getting
+        val iosSimulatorArm64Main by getting
+        val iosMain by creating {
+            dependsOn(commonMain)
+
+            iosX64Main.dependsOn(this)
+            iosArm64Main.dependsOn(this)
+            iosSimulatorArm64Main.dependsOn(this)
+        }
+
+        val androidMain by getting {
+            dependsOn(commonMain)
+        }
+
+        // testing
+        val commonTest by getting {
+            dependencies {
+                implementation(kotlin("test"))
+                implementation(libs.org.jetbrains.kotlin.test.common)
+                implementation(libs.org.jetbrains.kotlin.test.annotations.common)
+            }
+        }
+        val androidTest by getting {
+            dependencies {
+                implementation(kotlin("test"))
+                implementation(libs.org.jetbrains.kotlin.test.junit)
+            }
+        }
+        val desktopTest by getting {
+            dependencies {
+                implementation(kotlin("test"))
+                implementation(libs.org.jetbrains.kotlin.test.junit)
+            }
+        }
+        val iosX64Test by getting
```

Comment:
I *_think_* you can remove all of these targets that have empty blocks . They're redundant and not needed. Keeps the file cleaner/less polluted that way